### PR TITLE
revert back --skip-tls flag changes with deprecation notice

### PR DESCRIPTION
**Pull request by laxmikantbpandhare**
**Description of the change:**
Reverted changes for the `--skip-tls` flag and made sure that `--skip-tls-verify` marked true if the user enters `--skip-tls` with deprecation notice.

**Motivation for the change:**
We kept the `skip-tls` flag with operator SDK so that the user can use it and get notified via deprecation notice.

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [x] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Comment by laxmikantbpandhare** [comment]
@jmrodri - I added code change with deprecation notice. Do we need to add notification on the website docs as well?

**Review by jmrodri — CHANGES_REQUESTED** [review]
I would not have added SkipTLS as parameters to LoadBundle or anywhere that far. What I think we should do is have the flag be on the CLI (marked as deprecated). When the user passes that flag in we should, as early as possible, capture that they set it. THEN set SkipTLSVerify = true and UseHTTP = true. 

The code after the CLI parsing should only ever care about SkipTLSVerify and UseHTTP. It should no longer know what SkipTLS means. 

The reason SkipTLS was removed from OPM was because if it was true, it also reverted to using http. They did not want it to behave that way. That's why it was split into two flags. So in order to simulate what SkipTLS used to do is to set both of the new flags to true.

File: changelog/fragments/revert-skiptls-changes-deprc-notice.yaml
Comment (jmrodri) [inline changelog/fragments/revert-skiptls-changes-deprc-notice.yaml:3]:
```suggestion
      Reverted removal of  `--skip-tls` flag and mark the flag as deprecated.
```

File: internal/cmd/operator-sdk/scorecard/cmd.go
Comment (jmrodri) [inline internal/cmd/operator-sdk/scorecard/cmd.go:276]: I would not add SkipTLS to this call.

File: internal/olm/operator/bundle/install.go
Comment (jmrodri) [inline internal/olm/operator/bundle/install.go:75]: I would not pass SkipTLS to this call. Instead I would pass in true for both SkipTLSVerify AND UseHTTP *IF* SkipTLS was set.

File: internal/olm/operator/bundleupgrade/upgrade.go
Comment (jmrodri) [inline internal/olm/operator/bundleupgrade/upgrade.go:72]: do not pass SkipTLS to this

File: internal/olm/operator/helpers.go
Comment (jmrodri) [inline internal/olm/operator/helpers.go:37]: no need for skipTLS on this call.

File: internal/registry/image.go
Comment (jmrodri) [inline internal/registry/image.go:57]: This check should be done much earlier in the process, closer to where cobra is parsing the flag.

**Comment by asmacdo** [comment]
If we release without this, it would have a backwards incompatible change, so we need to block the 1.19 release until this is merged. Please ping @theishshah when this merges.

File: internal/registry/image.go
Comment (laxmikantbpandhare) [reply internal/registry/image.go:57]: Removed it.

File: internal/olm/operator/helpers.go
Comment (laxmikantbpandhare) [reply internal/olm/operator/helpers.go:37]: Removed it.

File: internal/olm/operator/bundleupgrade/upgrade.go
Comment (laxmikantbpandhare) [reply internal/olm/operator/bundleupgrade/upgrade.go:72]: Removed it.

File: internal/olm/operator/bundle/install.go
Comment (laxmikantbpandhare) [reply internal/olm/operator/bundle/install.go:75]: Removed from it.

File: changelog/fragments/revert-skiptls-changes-deprc-notice.yaml
Comment (laxmikantbpandhare) [reply changelog/fragments/revert-skiptls-changes-deprc-notice.yaml:3]: Modified.

**Comment by jmrodri** [comment]
> Sure, I updated the code and made sure that LoadBundle will not have SkipTLS flag as a parameter. One quick question, I tried to set both skip-tls-verify and use-http flags but run bundle is failing with below error Error: invalid flag combination: --use-http and --skip-tls-verify cannot both be true. That's why I set only skip-tls-verify to true if the user enters skip-tls flag thru the command line. Does this change sound good?
@laxmikantbpandhare I'll look at what those flags mean on the `opm` side.

**Comment by laxmikantbpandhare** [comment]
> > Sure, I updated the code and made sure that LoadBundle will not have SkipTLS flag as a parameter. One quick question, I tried to set both skip-tls-verify and use-http flags but run bundle is failing with below error Error: invalid flag combination: --use-http and --skip-tls-verify cannot both be true. That's why I set only skip-tls-verify to true if the user enters skip-tls flag thru the command line. Does this change sound good?
> 
> @laxmikantbpandhare I'll look at what those flags mean on the `opm` side.
> 
> Okay, `opm` uses `use-http` when `skip-tls` is being passed in. So that's what we want to do instead. https://github.com/operator-framework/operator-registry/blob/master/cmd/opm/internal/util/util.go#L34-L37

That's great. Modified the code accordingly.

File: internal/olm/operator/bundle/install.go
Comment (jmrodri) [inline internal/olm/operator/bundle/install.go:74]: I think comment is now out of date. 
```suggestion
	/ /if user sets --skip-tls then set --use-http to true as --skip-tls is deprecated
```

File: internal/olm/operator/registry/index/registry_pod.go
Comment (jmrodri) [inline internal/olm/operator/registry/index/registry_pod.go:80]:
```suggestion
	// SkipTLS controls whether to ignore SSL errors while pulling bundle image from registry server.
```

**Review by jmrodri — CHANGES_REQUESTED** [review]
a couple typos in the comments.

File: internal/olm/operator/registry/index/registry_pod.go
Comment (laxmikantbpandhare) [reply internal/olm/operator/registry/index/registry_pod.go:80]: Updated.

File: internal/olm/operator/bundle/install.go
Comment (laxmikantbpandhare) [reply internal/olm/operator/bundle/install.go:74]: Updated

**Comment by laxmikantbpandhare** [comment]
> a couple typos in the comments.

Updated typos.

**Review by jmrodri — APPROVED** [review]
/lgtm

**Review by rashmigottipati — CHANGES_REQUESTED** [review]
`skipTLS` field is added to registry pod struct but is not being used in the file. It should be handled in the the container creation command.

**Comment by rashmigottipati** [comment]
You need to add `--skip-tls={{ $.SkipTLS }}` to `opm registry add` command during container creation.

**Comment by laxmikantbpandhare** [comment]
> You need to add `--skip-tls={{ $.SkipTLS }}` to `opm registry add` command during container creation.

@rashmigottipati - Is there any need for this as we are setting `--use-http` flag once we encounter `--skip-tls` flag.

File: internal/olm/operator/registry/index/registry_pod.go
Comment (rashmigottipati) [inline internal/olm/operator/registry/index/registry_pod.go:80]:
```suggestion
```

Since this is not being used in registry pod creation anymore.

**Comment by rashmigottipati** [comment]
I looked into the latest opm registry add command and skipTLS flag is removed. Since we are defaulting to use `--use-http` flag when we encounter --skip-tls and also marking it as deprecated, I'm good with the changes. 

/hold remove

**Review by rashmigottipati — APPROVED** [review]

File: internal/olm/operator/registry/index/registry_pod.go
Comment (laxmikantbpandhare) [reply internal/olm/operator/registry/index/registry_pod.go]: removed this comment.

File: internal/olm/operator/registry/index/registry_pod.go
Comment (rashmigottipati) [inline internal/olm/operator/registry/index/registry_pod.go]:
```suggestion
```

Comment (laxmikantbpandhare) [reply internal/olm/operator/registry/index/registry_pod.go]: removed it.

**Review by rashmigottipati — APPROVED** [review]
/lgtm

**Comment by laxmikantbpandhare** [comment]
Removed `release-blocker` tag as it got merged now.

**Pull request merged**
